bb/sbuf2.c

```diff
@@ -716,35 +719,50 @@ void SBUF2_FUNC(sbuf2setflags)(SBUF2 *sb, int flags)
 
 SBUF2 *SBUF2_FUNC(sbuf2open)(int fd, int flags)
 {
+    if (fd < 0) {
+        return NULL;
+    }
     SBUF2 *sb = NULL;
-
-    /* Check for 'valid' file descriptor */
-    if (fd >= 0) {
-        sb = (SBUF2 *)calloc(1, sizeof(SBUF2));
-        if (!sb)
-            return NULL;
-        sb->fd = fd;
-        sb->flags = flags;
+#if SBUF2_SERVER
+    comdb2ma alloc = comdb2ma_create(0, 0, "sbuf2", 0);
+    if (alloc == NULL) {
+        goto error;
+    }
+    /* get malloc to work in server-mode */
+    SBUF2 dummy = {.allocator = alloc};
+    sb = &dummy;
```

Comment:
In server mode malloc -> comdb2_malloc(sb->allocator,...)
We create allocator first, then allocate sb from it. This sets up pointer to make above call work.